File: java/src/sample/replicant/sample/MockWidgetPainter.java
// Copyright 2009 Kiel Hodges
package replicant.sample;

import replicant.*;
import static replicant.Replicant.*;

public class MockWidgetPainter implements WidgetPainter {

  public void paint(Widget widget) {
    paint.call(widget);
  }
  
  public final MethodMocker1<Void, Widget> paint = voidMethodMocker(this, "paint").mappingResponses().with1Arg();
  
}

File: scala/src/test/scala/sample/MockWidgetPainter.scala
// Copyright 2011 Kiel Hodges
package sample

import replicant._

class MockWidgetPainter extends MockMinder[WidgetPainter] { minder =>

  val mock: WidgetPainter = new WidgetPainter {
    def paintWidget(widget: Widget) { minder.paintWidget(widget) }
  }

  val paintWidget = method("paintWidget", mock.paintWidget _)

}
